Client/BraveSkus/BraveSkusWebHelper.swift

```diff
@@ -102,18 +103,23 @@ class BraveSkusWebHelper {
   }
   
   /// Takes credential passed from the Brave SKUs and extract a proper credential to pass to the GuardianConnect framework.
-  static func fetchVPNCredential(_ credential: String, domain: String) -> (credential: String, environment: String)? {
+  static func fetchVPNCredential(_ credential: String, domain: String) -> SkusVPNCredential? {
     guard let unescapedCredential = credential.unescape(),
           let env = environment(domain: domain),
             let sampleUrl = URL(string: "https://brave.com") else { return nil }
     
-    guard let guardianConnectCredential =
-      HTTPCookie.cookies(withResponseHeaderFields:
-                          ["Set-Cookie": unescapedCredential], for: sampleUrl).first?.value else {
+    guard let cookie = HTTPCookie.cookies(withResponseHeaderFields:
+                                            ["Set-Cookie": unescapedCredential], for: sampleUrl).first else {
       return nil
     }
     
-    return (guardianConnectCredential, env)
+    let guardianCredential = cookie.value
+    
+    guard let expirationDate = cookie.expiresDate else {
+      return nil
```

Comment:
Fetching expiration date from the cookie

Reply:
I think we need to set the preferences too - does that happen in init?

Reply:
persisting is done in `prepareCredentialsPresentation` right after we cal this `fetchVPNCredential` method.
This method is only to parse the cookie, perhaps naming is not ideal